internal/api/api.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/btp-manager/internal/api/vm"
+	clusterobject "github.com/kyma-project/btp-manager/internal/cluster-object"
+	servicemanager "github.com/kyma-project/btp-manager/internal/service-manager"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type API struct {
+	cluster                 *client.Client
+	namespaceProvider       *clusterobject.NamespaceProvider
+	serviceInstanceProvider *clusterobject.ServiceInstanceProvider
+	secretProvider          *clusterobject.SecretProvider
+	serviceManager          *servicemanager.Client
+}
+
+func NewAPI() (*API, error) {
+	k8sCfg, err := config.GetConfig()
+	if err != nil {
+		panic(err)
+	}
+	k8s, err := client.New(k8sCfg, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+	log := slog.New(slog.NewTextHandler(os.Stdout, nil))
+	namespaceProvider := clusterobject.NewNamespaceProvider(k8s, log)
+	serviceInstanceProvider := clusterobject.NewServiceInstanceProvider(k8s, log)
+	secretProvider := clusterobject.NewSecretProvider(k8s, namespaceProvider, serviceInstanceProvider, log)
+	serviceManager := servicemanager.NewClient(context.Background(), log, secretProvider)
+
+	return &API{
+		cluster:                 &k8s,
+		namespaceProvider:       namespaceProvider,
+		serviceInstanceProvider: serviceInstanceProvider,
+		secretProvider:          secretProvider,
+		serviceManager:          serviceManager,
+	}, nil
```

Comment:
I think that all the api needs is SecretProvider and SM client which can be created in the main func and then passed to the constructor.